numpy/polynomial/polyutils.py

```diff
@@ -690,6 +690,280 @@ def _fit(vander_f, x, y, deg, rcond=None, full=False, w=None):
     else:
         return c
 
+def _get_coeff_idx(shape):
+    """
+    Get the powers of x and y corresponding to a 2d coefficient matrix
+    in the same order as the vandermode matrix,
+    i.e. [[0, 0], [1, 0], ...]
+
+    Parameters
+    ----------
+    shape : 2-tuple
+        shape of the coefficient matrix, i.e. degree + 1
+
+    Returns
+    -------
+    idx : array of shape (n * m, 2)
+        pairs of indices to the coefficient matrix
+    """
+    idx = np.indices(shape)
```

Comment:
Can you add `assert len(shape) == 2` here?

Reply:
Done